src/System.Management.Automation/engine/SessionStateContainer.cs

```diff
@@ -1648,7 +1648,25 @@ internal Collection<PSObject> GetChildItems(string[] paths, bool recurse, uint d
 
                 ContainerCmdletProvider providerInstance = GetContainerProviderInstance(provider);
 
-                if (path != null && this.ItemExists(providerInstance, path, context))
+                if (
+                    (context.Include != null && context.Include.Count > 0) ||
+                    (context.Exclude != null && context.Exclude.Count > 0))
+                {
+                    // Do the recursion manually so that we can apply the
+                    // include and exclude filters
```

Comment:
Should we check "context.Stopping" as in line 1587?

Reply:
This is not necessary.  In the previous code, if a user try to cancel the search in between, the code will stop on next path search iteration. In this situation, since we only have one literal path, it won't work.

Reply:
Thanks for clarify!

Closed.